events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+	return JANUS_EVENTHANDLER_API_VERSION;
+}
+
+int janus_gelfevh_get_version(void) {
+	return JANUS_GELFEVH_VERSION;
+}
+
+const char *janus_gelfevh_get_version_string(void) {
+	return JANUS_GELFEVH_VERSION_STRING;
+}
+
+const char *janus_gelfevh_get_description(void) {
+	return JANUS_GELFEVH_DESCRIPTION;
+}
+
+const char *janus_gelfevh_get_name(void) {
+	return JANUS_GELFEVH_NAME;
+}
+
+const char *janus_gelfevh_get_author(void) {
+	return JANUS_GELFEVH_AUTHOR;
+}
+
+const char *janus_gelfevh_get_package(void) {
+	return JANUS_GELFEVH_PACKAGE;
+}
+
+void janus_gelfevh_incoming_event(json_t *event) {
+	if(g_atomic_int_get(&stopping) || !g_atomic_int_get(&initialized)) {
+		/* Janus is closing or the plugin is */
+		return;
+	}
+
+	/* Do NOT handle the event here in this callback! Since Janus notifies you right
+	 * away when something happens, these events are triggered from working threads and
+	 * not some sort of message bus. As such, performing I/O or network operations in
+	 * here could dangerously slow Janus down. Let's just reference and enqueue the event,
+	 * and handle it in our own thread: the event contains a monotonic time indicator of
+	 * when the event actually happened on this machine, so that, if relevant, we can compute
+	 * any delay in the actual event processing ourselves. */
+	json_incref(event);
+	//JANUS_LOG(LOG_WARN, "Got event: %s\n", json_dumps(event, json_format));
+	g_async_queue_push(events, event);
+
+}
+
+json_t *janus_gelfevh_handle_request(json_t *request) {
+	if(g_atomic_int_get(&stopping) || !g_atomic_int_get(&initialized)) {
+		return NULL;
+	}
+	/* We can use this requests to apply tweaks to the logic */
+	int error_code = 0;
+	char error_cause[512];
+	JANUS_VALIDATE_JSON_OBJECT(request, request_parameters,
+		error_code, error_cause, TRUE,
+		JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+	if(error_code != 0)
+		goto plugin_response;
+	/* Get the request */
+	const char *request_text = json_string_value(json_object_get(request, "request"));
+	if(!strcasecmp(request_text, "tweak")) {
+		/* We only support a request to tweak the current settings */
+		JANUS_VALIDATE_JSON_OBJECT(request, tweak_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto plugin_response;
+		/* Parameters we can change */
+		const char *req_events = NULL, *req_backend = NULL, *req_port = NULL;
+		/* Events */
+		if(json_object_get(request, "events"))
+			req_events = json_string_value(json_object_get(request, "events"));
+		/* Backend stuff */
+		if(json_object_get(request, "backend"))
+			req_backend = json_string_value(json_object_get(request, "backend"));
+		if(json_object_get(request, "port"))
+			req_port = json_string_value(json_object_get(request, "port"));
+		if(req_backend && req_port) {
+			/* Invalid backend address, port */
```

Comment:
Not sure why this is a failed check..? Did you mean to place a `!` before one of the two?

Reply:
This hasn't been addressed yet: can you please clarify what this check is for?